flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -350,6 +359,10 @@ private long uploadSnapshotFiles(
                                 ? CheckpointedStateScope.EXCLUSIVE
                                 : CheckpointedStateScope.SHARED;
 
+                if (stateFileVerifier != null) {
+                    stateFileVerifier.verifySstFilesChecksum(sstFilePaths);
+                }
+
                 List<HandleAndLocalPath> sstFilesUploadResult =
                         stateUploader.uploadFilesToCheckpointFs(
```

Comment:
From the code order, we verify the file first, and then `uploadFilesToCheckpointFs`.

If the file is corrupted before calling `hdfs.write()`, this check still cannot cover, right?

Reply:
> From the code order, we verify the file first, and then `uploadFilesToCheckpointFs`.
> 
> If the file is corrupted before calling `hdfs.write()`, this check still cannot cover, right?


Yes ,  If the file is corrupted before calling `hdfs.write()`, this check still cannot cover

But as I understand,  After the file is hardlinked to snapshotDir, no other operations will change the file. So there should be no file corruption during this process.

I think there are only two places where corruption may occur, one is when the file is created, and the other is when it is re-downloaded from Checkpoint to the local disk.

Reply:
> most files are damaged due to hardware failures on the machine where the file is written

Check with you first, if there is hardware failures after the file is written,, doesn't it happen?

Reply:
> > most files are damaged due to hardware failures on the machine where the file is written
> 
> Check with you first, if there is hardware failures after the file is written,, doesn't this issue happen?

Well, just like I replied to you in the ticket, I think this possibility is very small. We can continue the discussion in the ticket and then come back to continue reviewing this PR.